src/rust/ensogl/lib/components/src/scrollbar.rs

```diff
@@ -0,0 +1,267 @@
+                        0.0
+                    } else if active || dragging {
+                        1.0
+                    } else {
+                        if dist <= 0.0 {
+                            1.0
+                        } else {
+                            (0.7 - dist / 20.0).max(0.0).min(1.0)
+                        }
+                    }
+                });
+
+
+            // === Position on Screen ===
+
+            // Space that the thumb can actually move in
+            inner_length        <- frp.set_length.map(|length| *length - 2.0 * PADDING);
+            // Thumb position as a number between 0 and 1
+            normalized_position <- all_with3(&frp.thumb_position,&frp.set_thumb_size,&frp.set_max,
+                |&pos,&size,&max| pos / (max - size));
+            normalized_size     <- all_with(&frp.set_thumb_size,&frp.set_max,|&size,&max|
+                size / max);
+            // Minimum thumb size in normalized units
+            min_visual_size     <- inner_length.map(|&length| MIN_THUMB_SIZE / length);
+            // The size at which we render the thumb on screen, in normalized units. Can differ from
+            // the actual thumb size if the thumb is smaller than the min.
+            visual_size         <- all_with(&normalized_size,&min_visual_size,|&size,&min|
+                size.max(min).min(1.0));
+            // The position at which we render the thumb on screen, in normalized units.
+            visual_start        <- all_with(&normalized_position,&visual_size,|&pos,&size|
+                pos * (1.0 - size));
+            visual_bounds       <- all_with(&visual_start,&visual_size,|&start,&size|
+                Bounds::new(start,start+size));
+            visual_center       <- visual_bounds.map(|bounds| bounds.center());
+            thumb_center_px     <- all_with(&visual_center,&inner_length, |normalized,length|
+                (normalized - 0.5) * length);
+
+            update_slider <- all(&visual_bounds,&resize);
+            eval update_slider(((value,size)) model.set_background_range(*value,*size));
+
+
+            // === Clicking ===
+
+            frp.scroll_by <+ base_frp.background_click.map3(&thumb_center_px,&frp.set_thumb_size,
+                |click_position,thumb_center,thumb_size| {
+                    let direction = if click_position.x > *thumb_center { 1.0 } else { -1.0 };
+                    direction * thumb_size * CLICK_JUMP_PERCENTAGE
+                });
+
+
+            // === Dragging ===
+
+            drag_started <- base_frp.is_dragging_track.on_change().on_true().constant(());
+            drag_offset  <- all4(&mouse_position,&inner_length,&frp.set_max,&frp.thumb_position)
+                .sample(&drag_started)
+                .map(|(mouse_px,length_px,max,thumb_pos)| {
+                    let thumb_position_px = thumb_pos / max * length_px;
+                    mouse_px.x - thumb_position_px
+                });
```

Comment:
according to our coding guidelines, the chain should be split to many `<-` instructions.